tests/integration/src/xvm.rs

```diff
@@ -0,0 +1,334 @@
+            hex::decode("3ccfd60b").expect("invalid selector hex"),
+            0,
+        ));
+        assert_eq!(
+            Balances::free_balance(account_id_from(evm_payable_addr)),
+            ExistentialDeposit::get(),
+        );
+    });
+}
+
+#[test]
+fn wasm_payable_call_via_xvm_works() {
+    new_test_ext().execute_with(|| {
+        let contract_addr = deploy_wasm_contract("payable");
+
+        let prev_balance = Balances::free_balance(&contract_addr);
+        let value = UNIT;
+        assert_ok!(Xvm::call(
+            Context {
+                source_vm_id: VmId::Evm,
+                weight_limit: Weight::from_parts(10_000_000_000, 1024 * 1024),
+            },
+            VmId::Wasm,
+            ALICE,
+            MultiAddress::<AccountId32, ()>::Id(contract_addr.clone()).encode(),
+            // Calling `deposit`
+            hex::decode("0000002a").expect("invalid selector hex"),
+            value
+        ));
+        assert_eq!(
+            Balances::free_balance(contract_addr.clone()),
+            value + prev_balance
+        );
+    });
+}
+
+#[test]
+fn calling_wasm_payable_from_evm_fails_if_caller_contract_balance_below_ed() {
+    new_test_ext().execute_with(|| {
+        let wasm_payable_addr = deploy_wasm_contract("payable");
+        let call_wasm_payable_addr = deploy_evm_contract(CALL_WASM_PAYBLE);
+
+        let value = 1_000_000_000;
+        assert_ok!(EVM::call(
+            RuntimeOrigin::root(),
+            alith(),
+            call_wasm_payable_addr.clone(),
+            // to: 0x00a8f69d59df362b69a8d4acdb9001eb3e1b8d067b8fdaa70081aed945bde5c48c
+            // input: 0x0000002a (deposit)
+            // value: 1000000000
+            hex::decode("4012b914000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000c0000000000000000000000000000000000000000000000000000000003b9aca00000000000000000000000000000000000000000000000000000000000000002100a8f69d59df362b69a8d4acdb9001eb3e1b8d067b8fdaa70081aed945bde5c48c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000040000002a00000000000000000000000000000000000000000000000000000000").expect("invalid call input hex"),
+            U256::from(value),
+            1_000_000,
+            U256::from(DefaultBaseFeePerGas::get()),
+            None,
+            None,
+            vec![],
+        ));
+
+        // TODO: after XVM error propagation finished, assert `pallet-evm` execution error
+        // and update balance assertions.
+
+        // Transfer to EVM contract ok.
+        assert_eq!(
+            Balances::free_balance(&account_id_from(call_wasm_payable_addr)),
+            value,
+        );
+        // Transfer from EVM contract to wasm Contract err.
+        assert_eq!(
+            Balances::free_balance(&wasm_payable_addr),
+            ExistentialDeposit::get(),
+        );
+    });
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The test function `calling_wasm_payable_from_evm_fails_if_caller_contract_balance_below_ed` tests the scenario where the caller contract's balance is below the existential deposit (ED). The test deploys a Wasm contract and an EVM contract, attempts to perform a payable call from the EVM contract to the Wasm contract, and verifies that the transfer fails due to insufficient balance. No issues found.



<!-- This is an auto-generated comment by OSS CodeRabbit -->